web/src/message_lists.ts

```diff
@@ -16,6 +16,7 @@ type MessageListView = {
     _render_win_end: number;
 };
 
+export type MessageListAddMessagesReturn = undefined | {need_user_to_scroll: boolean};
```

Comment:
Maybe `RenderInfo` to reuse the existing name? And leave `undefined` out of this definition (so affected functions return `RenderInfo | undefined`.)

Reply:
Good ideas, made those changes